### ISSUE TITLE
Windows 10 pro x64, Git 2.20.1 Windows x64, Unable to edit C:\Program Files\Git\git-bash.exe (out of memory).

### ISSUE THREAD

**Issue by donhuvy** [git-for-windows/git#2003]
Windows 10 pro x64.

Git for Windows version 2.20.1 x64.

I try to reinstall, restart many time, Run as administrator, but not success.

<img width="374" alt="git" src="https://user-images.githubusercontent.com/1328316/50554858-204dc800-0cf5-11e9-9765-7234329e00d9.png">

also version 2.20.1 for Windows 32 bit, 64 bit cause same error (like screenshot).

**Comment by donhuvy** [comment]
```
C:\Users\donhuvy>git --version --build-options
git version 2.20.0.windows.1
cpu: x86_64
built from commit: 95155834166f64fe9666f2c0a4909f076080893a
sizeof-long: 4
sizeof-size_t: 8

C:\Users\donhuvy>
```

**Comment by donhuvy** [comment]
```
C:\Users\donhuvy>cmd.exe /c ver

Microsoft Windows [Version 10.0.16299.847]

C:\Users\donhuvy>
```

**Comment by donhuvy** [comment]
Install by file `Git-2.20.0-64-bit.exe`

**Comment by dscho** [comment]
That sounds as if you tried to use Windows' default console window for Git Bash (note: the issue reporting template *really* is not meant to annoy bug reporters, but rather intended to help you not forget vital information).

Can you edit `git-bash.exe` via [ResEdit](http://www.resedit.net/)? Its first string table has two entries, and you would want to remove the first one.

**Comment by donhuvy** [comment]
- I used previous version of Git for Windows, and the problem did not exist. 

- You said true, I tried to use Windows' default console (cmd).

- A warning popup display, it is wrong. It is exactly a bug.

**Comment by dscho** [comment]
@donhuvy please do try to work with me, if you do not try ResEdit I get the impression that you might not be interested in resolving this bug at all.

**Comment by donhuvy** [comment]
ok, I will reply soon. I am using macOS, I will report about it when use Windows OS.

**Comment by donhuvy** [comment]
This is my result

<img width="800" alt="a1" src="https://user-images.githubusercontent.com/1328316/50723622-c7c26480-1112-11e9-861f-ba4bdc61e00f.png">
<img width="800" alt="a2" src="https://user-images.githubusercontent.com/1328316/50723623-c85afb00-1112-11e9-8836-e153bb7a69e1.png">

**Comment by dscho** [comment]
On Sat, 5 Jan 2019, Do Nhu Vy wrote:

> This is my result
> 
> <img width="800" alt="a1" src="https://user-images.githubusercontent.com/1328316/50723622-c7c26480-1112-11e9-861f-ba4bdc61e00f.png">
> <img width="800" alt="a2" src="https://user-images.githubusercontent.com/1328316/50723623-c85afb00-1112-11e9-8836-e153bb7a69e1.png">

Could you summarize that without pasting an image that I cannot see in a
text reader?

**Comment by donhuvy** [comment]
The content of it is 

APP_ID=GitForWindows.Bash ALLOC_CONSOLE=1 usr\\bin\\mintty.exe --nodaemon -o AppID=GitForWindows.Bash -o AppLaunchCmd=\"@@EXEPATH@@\\git-bash.exe\" -o AppName=\"Git Bash\" -i \"@@EXEPATH@@\\git-bash.exe\" --store-taskbar-properties -- /usr/bin/bash --login -i

**Comment by alegrigoriev** [comment]
What antivirus do you have?

**Comment by donhuvy** [comment]
Yes, I use McAfee Antivirus

<img width="659" alt="mcafee" src="https://user-images.githubusercontent.com/1328316/51095172-530fba00-17e5-11e9-919f-ca5606af3031.PNG">

**Comment by donhuvy** [comment]
After I use ResEdit on file `git-bash.exe`, I have what I expected.

**Comment by dscho** [comment]
I think there is a bug where the actual condition of the error is not handed through correctly, and the installer will always claim that it is out of memory, even if it is a matter of access being denied.

I just kicked off a snapshot build that should address the incorrect error message, and would like to ask you, @donhuvy, to test this installer; It should be available [here](https://wingit.blob.core.windows.net/files/index.html) once [the build](https://dev.azure.com/git-for-windows/git/_build/results?buildId=31514) is done.

**Comment by alegrigoriev** [comment]
To work around an antivirus, perhaps the installer needs to patch the file while in a temporary directory, and then copy the patched file to ProgramFiles/Git

**Comment by donhuvy** [comment]
I download Git for Windows 64bit - (commit 1dfd1896e4). Turn on Real-time protection of Mc-Afee antivirus.

![image](https://user-images.githubusercontent.com/1328316/53223902-7875ca80-36a5-11e9-89fe-c54caad46e36.png)

<img width="374" alt="git1" src="https://user-images.githubusercontent.com/1328316/53223972-a8bd6900-36a5-11e9-8e0c-a5ec226662d6.PNG">

**Comment by dscho** [comment]
> To work around an antivirus, perhaps the installer needs to patch the file while in a temporary directory, and then copy the patched file to ProgramFiles/Git

That's a good suggestion. Could you open a PR to make it so?

**Comment by mfriedrich74** [comment]
A temp folder is probably inappropriate. The file should be on the same
filesystem/volume and moved over for performance reasons. But when you look
closer, I am pretty sure inno setup does that already.

Best regards, Mike

**Comment by dscho** [comment]
@mfriedrich74 you don't need to guess, you can look!

In any case, we already work with a temporary directory in similar cases, look for `ExpandConstant('{tmp}')` in [`install.iss`](https://github.com/git-for-windows/build-extra/blob/master/installer/install.iss).

To test out your changes:
1. Install [Git for Windows' SDK](https://gitforwindows.org/#download-sdk)
2. `sdk cd build-extra`
3. Edit `installer/install.iss` (look for "edit-git-bash", and use `CopyFile()` with the third parameter set to `True` to force-overwrite)
4. Test by building a test installer (`./installer/release.sh 0-test`) and running it.

**Comment by dscho** [comment]
Okay, people, I really described in detail how you can help yourself with this issue. Ball's in your court now.

**Comment by donhuvy** [comment]
I install `git version 2.21.0.windows.1` , it is ok.
